lib/dialects/abstract/query-generator.js

```diff
@@ -1135,14 +1135,19 @@ class QueryGenerator {
     attributes.main = this.escapeAttributes(attributes.main, options, mainTable.as);
     attributes.main = attributes.main || (options.include ? [`${mainTable.as}.*`] : ['*']);
 
-    // If subquery, we ad the mainAttributes to the subQuery and set the mainAttributes to select * from subquery
+    // If subquery, we add the mainAttributes to the subQuery and set the mainAttributes to select * from subquery
     if (subQuery || options.groupedLimit) {
       // We need primary keys
       attributes.subQuery = attributes.main;
       attributes.main = [`${mainTable.as || mainTable.quotedName}.*`];
     }
 
     if (options.include) {
+      // To support cross database joins on mysql, prefix the database name of the model in case the model is for a different database
+      if (this.dialect === 'mysql') {
```

Comment:
This is a rather crude approach, how will this API work? Can you show an example of include that you can use to cross join with current proposed changes?

Reply:
I agree it's a bit crude and seems hacky but I don't see another way of doing this reliably without affecting a lot of the apis and breaking existing features.

In an app I'm working on that uses multiple databases, I have multiple sequelize instances like below:

```
const Sequelize = require('Sequelize');
const userDb = new Sequelize(/* ... */);
const contentDb = new Sequelize(/* ... */);
// more databases here....
```

Then, I have several models for each database that are related to each other such as `userDb.ClientModel`, `contentDb.PostModel` where each client can have multiple posts so the relationship is defined as below:

```
userDb.ClientModel.hasMany(contentDb.PostModel, {foreign_key: 'client_id', sourceKey: 'id', constraints: false});
contentDb.PostModel.belongsTo(userDb.ClientModel, {foreign_key: 'client_id`, targetKey: 'id', constraints: false});
```

Now if I want to query all posts belonging to a user with 
```
userDb.ClientModel.findAll({
  include: { model: contentDb.PostModel }, 
  limit: 5
});
```
it will **NOT** work since the query will try to look for the `PostModel` table in the `userDb` database. However, I made it work with my implementation of prefixing the table name with the db name. 

It only becomes an issue when a query includes joins or subqueries however, I'm not super confident if my fix will work for subqueries that run on another table but from my tests. it does work for join queries such as the above explained scenario.

*All the above code are somewhat pseudo code and not exactly as the actual code from my project since I'm not at liberty of sharing those code. So if something doesn't add up or not clear enough, feel free to let me know.*